### DISCUSSION TITLE
From what sources does Navit get traffic data?

### DISCUSSION THREAD

**Discussion by Korb** [navit-gps/navit#1280] [Q&A] [answered]
And does it only take into account current traffic, or does the route take into account the expected normal traffic congestion on that day of the week?

**Comment by mvglasow** [comment, marked as answer]
Traffic in Navit is currently supported on Android only. Traffic data is obtained from other apps using the [TraFF interface](https://traffxml.org). The data supplied is in principle entirely up to these applications and the sources they draw their data from.

The data format allows not only for traffic flow, but also other factors with an impact on travel times, such as closures and roadworks (which frequently come with speed limits). However, it depends on reports which pertain to a particular stretch of a particular road. Event codes such as “heavy traffic expected” are available in theory, but only if the source supports them. In practice, I am not aware of any source that does.

Where reports do not come with quantified traffic flow data (speed limit, speed of moving traffic and/or delays in minutes), Navit has a set of rules to derive the impact on travel times. For example, roadwork areas are assumed to have a speed limit which is lower than the usual limit for that stretch.

So far there are two apps which supply traffic data:
* RoadEagle, which gets data from the road authorities for Latvia, Lithuania and Poland. For Latvia and Poland, this is mostly predictable information (such as roadworks and planned closures), although at least Poland has in the past also reported some accidents. For Lithuania, the event reports may also include accidents; additionally, live traffic flow data is available for the national road network.
* Qz, which requires extra hardware (a TMC receiver connected via USB or Bluetooth) and retrieves live traffic data from radio stations in a number of states, mostly in Western Europe. In practice, this usually includes roadworks as well as accidents, but data quality may vary.

**Reply by Korb** [reply]
Thank you very much for your detailed answer!

> In practice, I am not aware of any source that does.

Yandex does this:

![Screen Shot 2024-11-14 at 23 27 53](https://github.com/user-attachments/assets/319cf5ce-1acf-440a-aa0a-835cf3c259b8)

Using [their API](https://yandex.com/dev/jsapi-v2-1/doc/en/v2-1/ref/reference/ITrafficProvider) in addition to the current one could improve navigation in the former Soviet Union countries where the company does business.
